### PR TITLE
Add a modules option to load the modules from an arbitrary directory

### PR DESCRIPTION
**Pull request by nrdmao**
This option provides a degree of flexibility to BESS installations where the location of bessd is decouple from the location of the module .so's. For example, for consistency with other system daemons, one might install bessd in /usr/bin but the module .so's in /usr/lib/bess/modules. Under such circumstances bessd may be started as such: bessd -modules /usr/lib/bess/modules

### CODE REVIEW

File: core/main.cc [hunk 1]
```diff
@@ -39,9 +39,9 @@ int main(int argc, char *argv[]) {
   bess::bessd::WritePidfile(pidfile_fd, getpid());
 
   // Load plugins
-  if (!bess::bessd::LoadPlugins(bess::bessd::GetCurrentDirectory() +
-                                "modules")) {
-    PLOG(FATAL) << "LoadPlugins() failed";
+  if (!bess::bessd::LoadPlugins(FLAGS_modules)) {
+    PLOG(FATAL) << "LoadPlugins() failed to load from directory: " <<
```
Comment (changlan) [inline core/main.cc]: indentation

File: core/opts.cc [hunk 2]
```diff
@@ -19,7 +20,8 @@ DEFINE_bool(s, false, "Show TC statistics every second");
 DEFINE_bool(d, false, "Run BESS in debug mode (with debug log messages)");
 DEFINE_bool(a, false, "Allow multiple instances");
 DEFINE_bool(no_huge, false, "Disable hugepages");
-
+DEFINE_string(modules, bess::bessd::GetCurrentDirectory() + "modules",
+	      "Load modules from the specified directory");
```
Comment (changlan) [inline core/opts.cc]: add an empty line after this.

File: core/main.cc [hunk 3]
```diff
@@ -39,9 +39,9 @@ int main(int argc, char *argv[]) {
...
+      FLAGS_modules;
```
Comment (changlan) [inline core/main.cc]: indentation

File: core/opts.cc (hunk 2 above)
Comment (nrdmao) [inline core/opts.cc]: OK, done.

File: core/main.cc (hunk 3 above)
Comment (nrdmao) [inline core/main.cc]: OK, done.

**Comment by changlan** [comment]
image

**Pull request merged**